### ISSUE TITLE
shellcheck fails to parse tests when the comparator is a variable. 

### ISSUE THREAD

**Issue by derekschrock** [koalaman/shellcheck#3002]
#### For bugs
- SC1073
- version: 0.10.0
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

v1=1
v2=2
[ "$v1" > "$v2" ] || echo no
[ "$v2" -gt "$v1" ] && echo yes

gt=">"
[ "$v1" $gt "$v2" ] || echo no
gt="-gt"
[ "$v2" $gt "$v1" ] && echo yes

```

#### Here's what shellcheck currently says:

In test.sh line 9:
[ "$v1" $gt "$v2" ] || echo no
^-- SC1073 (error): Couldn't parse this test expression. Fix to allow more checks.
                ^-- SC1072 (error): Expected test to end here (don't wrap commands in []/[[]]). Fix any mentioned problems and try again. 

#### Here's what I wanted or expected to see:

shellcheck to be able to handle the test where test is a var.

**Comment by kevinoid** [comment]
This seems very similar to #1645.  Possibly a duplicate?

**Comment by derekschrock** [comment]
Looks like a duplicate to me.
